File: src/main/java/gg/mineral/practice/managers/EventManager.java
package gg.mineral.practice.managers;

import gg.mineral.api.collection.GlueList;
import gg.mineral.practice.events.Event;
import lombok.Getter;

public class EventManager {
    @Getter
    static GlueList<Event> events = new GlueList<>();

    public static void registerEvent(Event event) {
        events.add(event);
    }

    public static void remove(Event event) {
        events.remove(event);
    }

    public static Event getEventByName(String s) {
        for (Event event : events) {
            if (event.getHost().equalsIgnoreCase(s)) {
                return event;
            }
        }
        return null;
    }
}

File: src/main/kotlin/gg/mineral/practice/managers/EventManager.kt
package gg.mineral.practice.managers

import gg.mineral.practice.events.Event
import it.unimi.dsi.fastutil.objects.Object2ObjectOpenHashMap

object EventManager {
    val events: MutableMap<String, Event> = Object2ObjectOpenHashMap()

    fun registerEvent(event: Event) {
        events[event.host] = event
    }

    fun remove(event: Event) = events.remove(event.host)

    fun getEventByName(name: String) = events[name]
}
